### PR TITLE
Issue #81: Date inputs should respect users date format

### PR DESCRIPTION
**Pull request by Shivam0710**
I have done changes as follows:

- Moved location question before birthday so we know what country the user and based on the country of I've maintained a function where we can add the value based on the country code.
- Added fields in geo-utils for dateFormat so that we can pick the dateFormat based on country code.
- Added a prop to DatePicker component that takes in the date format and parses the date accordingly and sets an error message if the existing if (inputData.isValid) check fails.
- Removed hard coded formats in DatePicker component and made them dynamic based on the dateFormat provided.

### CODE REVIEW

File: libs/design-system/src/lib/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -25,11 +25,41 @@ export interface DatePickerProps {
...
+        .map((keyword) => keyword.toUpperCase())
+}
+
+function getDateFormatPattern(dateFormat: string): string {
+    const dateComponents = dateFormat.split('/')
+    const pattern = dateComponents
+        .map((component) => {
+            if (component.includes('d')) {
+                return ' ## ' // Placeholder for day
+            } else if (component.includes('m') || component.includes('M')) {
```
Comment (tmyracle) [inline libs/design-system/src/lib/DatePicker/DatePicker.tsx:52]: This is looking great, thanks for taking this on! Just a few small questions:

- Do we need to check both upper and lower case here? If so, should we do it for day and year as well?
- In the getMaskArray could you use a regex to remove slashes and spaces in one go, then upper case and split to array?

Comment (Shivam0710) [reply libs/design-system/src/lib/DatePicker/DatePicker.tsx]: Concerning the initial item, we only need to address "M," but I will adjust all of them to perform case-insensitive checks for added safety. As for the second point, I will manage the modifications accordingly.

Comment (Shivam0710) [reply libs/design-system/src/lib/DatePicker/DatePicker.tsx]: I've addressed the comments. Kindly review the pull request and inform me if any further modifications are necessary.

File: libs/design-system/src/lib/DatePicker/DatePicker.tsx
Comment (tmyracle) [inline libs/design-system/src/lib/DatePicker/DatePicker.tsx:1]: After the user has set their country can we use that to set the format throughout the app. I think the changes so far only update it for the Profile step but everywhere else in the app will still be the default it was before.

Comment (Shivam0710) [reply libs/design-system/src/lib/DatePicker/DatePicker.tsx:1]: Yes the issue is related to Profile only but we can save the country code in the cookies and retrieve it when picking up the format. Can you please list down the scenarios where we'll have to take care of this? I'll handle this accordingly.

Comment (tmyracle) [reply libs/design-system/src/lib/DatePicker/DatePicker.tsx:1]: DatePicker is used in multiple places in the app like <UserDetails />, <ValuationTableForm />, etc. Can just do a search for all the uses of DatePicker.

Comment (Shivam0710) [reply libs/design-system/src/lib/DatePicker/DatePicker.tsx:1]: Closed this by mistake. I have opened PR for the same

**Pull request closed without merging**